presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowColumnStatsRewriteResultBuilder.java

```diff
@@ -0,0 +1,88 @@
+import com.facebook.presto.spi.statistics.ColumnStatistics;
+import com.facebook.presto.spi.statistics.Estimate;
+import com.facebook.presto.spi.statistics.TableStatistics;
+import com.facebook.presto.sql.QueryUtil;
+import com.facebook.presto.sql.tree.DoubleLiteral;
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.NullLiteral;
+import com.facebook.presto.sql.tree.Row;
+import com.facebook.presto.sql.tree.SelectItem;
+import com.facebook.presto.sql.tree.StringLiteral;
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+class ShowColumnStatsRewriteResultBuilder
+{
+    private ShowColumnStatsRewriteResultBuilder() {}
+
+    static List<Expression> buildStatisticsRows(TableStatistics tableStatistics, Set<String> statisticsNames)
+    {
+        ImmutableList.Builder<Expression> rowsBuilder = ImmutableList.builder();
+
+        // Stats for columns
+        for (Map.Entry<ColumnHandle, ColumnStatistics> columnStats : tableStatistics.getColumnStatistics().entrySet()) {
+            Map<String, Estimate> columnStatisticsValues = columnStats.getValue().getStatistics();
+            rowsBuilder.add(createStatsRow(Optional.of(columnStats.getKey().toString()), statisticsNames, columnStatisticsValues));
+        }
+
+        // Stats for whole table
+        rowsBuilder.add(createStatsRow(Optional.empty(), statisticsNames, tableStatistics.getTableStatistics()));
+
+        return rowsBuilder.build();
+    }
+
+    static SelectItem[] buildSelectItems(List<String> columnNames)
+    {
+        return columnNames.stream().map(QueryUtil::unaliasedName).toArray(SelectItem[]::new);
+    }
+
+    static List<String> buildColumnsNames(Set<String> statisticsNames)
+    {
+        ImmutableList.Builder<String> columnNamesBuilder = ImmutableList.builder();
+        columnNamesBuilder.add("column_name");
+        columnNamesBuilder.addAll(statisticsNames);
+        return columnNamesBuilder.build();
+    }
+
+    private static Row createStatsRow(Optional<String> columnName, Set<String> statisticsNames, Map<String, Estimate> columnStatisticsValues)
+    {
+        ImmutableList.Builder<Expression> rowValues = ImmutableList.builder();
+        Expression columnNameExpression = columnName.map(name -> (Expression) new StringLiteral(name)).orElse(new NullLiteral());
+
+        rowValues.add(columnNameExpression);
+        for (String statName : statisticsNames) {
+            rowValues.add(createStatisticValueOrNull(columnStatisticsValues, statName));
+        }
+        return new Row(rowValues.build());
+    }
+
+    private static Expression createStatisticValueOrNull(Map<String, Estimate> columnStatisticsValues, String statName)
+    {
+        if (columnStatisticsValues.containsKey(statName)) {
+            return new DoubleLiteral(Double.toString(columnStatisticsValues.get(statName).getValue()));
```

Comment:
https://www.youtube.com/watch?v=yqyixwqiCag

If you think that they are integers why don't you keep them as integers in the first place? IMHO changing the type in here will sooner or later cause some debugging problems as we will not display what is actually used across the code.